Svc/TraceFileLogger/TraceFileLogger.cpp

```diff
@@ -0,0 +1,238 @@
+// ======================================================================
+// \title  TraceFileLogger.cpp
+// \author sreddy
+// \brief  cpp file for TraceFileLogger component implementation class
+// ======================================================================
+
+#include "Svc/TraceFileLogger/TraceFileLogger.hpp"
+#include "FpConfig.hpp"
+#include <Fw/Types/Assert.hpp>
+#include <Os/File.hpp>
+#include <Os/FileSystem.hpp>
+#include <limits>
+#include <cstring>
+#include <cstdio>
+#include <Fw/Types/StringUtils.hpp>
+#include <Fw/Types/BasicTypes.hpp>
+#include <Fw/Trace/TraceEntry.hpp>
+
+namespace Svc {
+
+    // ----------------------------------------------------------------------
+    // Component construction and destruction
+    // ----------------------------------------------------------------------
+
+    TraceFileLogger::TraceFileLogger(const char* const compName) : 
+    TraceFileLoggerComponentBase(compName),
+    m_log_file(),
+    m_mode(CLOSED),
+    m_fileName(),
+    m_maxFileSize(0),
+    m_byteCount(0),
+    m_log_init(false),
+    m_traceFilter(0)
```

Comment:
TODO: Make an array proc constructor to pass the array reference and initialize the reference to the array here.